src/graphql/resolvers/index.js

```diff
@@ -168,49 +169,85 @@ module.exports = function Resolvers (ssb) {
   }
 
   /**
-   * Takes the messages from a thread and maps them based on whether
-   * a profile is returned for the author of that message or not.
-   * Messages from someone who hasnt yet opted in to publicWebHosting or
-   * a profile wasnt found for them, will return empty values
-   * @param {array} messages - messages in a thread
+   * Takes a messages and maps it either to a Comment or "empty" Comment depending on
+   * whether the author has opted into publicWebHosting.
+   * @param {object} msg - a post message in kv format
+   * @param {function} cb - an error first callback function
    */
-  const mapMessages = (messages) => {
-    return new Promise((resolve, reject) => {
-      pull(
-        pull.values(messages),
-        pullParaMap((msg, cb) => {
-          getProfile(msg.value.author)
-            .then(profile => {
-              // if there was a profile, thats great
-              if (profile) {
-                return cb(null, {
-                  id: msg.key,
-                  author: msg.value.author,
-                  text: msg.value.content.text,
-                  timestamp: msg.value.timestamp // asserted publish time
-                })
+  function publicifyMessage (msg, cb) {
```

Comment:
the `mapMessages` method name distracted me and I did this refactor. It should have been another PR